web/src/main/java/org/openmrs/web/Listener.java

```diff
@@ -167,13 +170,17 @@ public void sessionDestroyed(HttpSessionEvent se) {
 	 * @see HttpSessionListener#sessionDestroyed(HttpSessionEvent)
 	 */
 	private List<HttpSessionListener> getHttpSessionListeners() {
-		List<HttpSessionListener> httpSessionListeners = new ArrayList<>();
-		try {
-			httpSessionListeners = Context.getRegisteredComponents(HttpSessionListener.class);
-		}
-		catch (Exception e) {
-			log.warn("An error occurred trying to retrieve HttpSessionListener beans from the context", e);
+		List<HttpSessionListener> httpSessionListeners = Collections.emptyList();
+		
+		if (initialized) {
```

Comment:
Putting a check around this seems fine with me.  I don't know if this will miss any particular session events, but if we are relying on listeners registered in the application context, we don't really have any choice but to wait for the application context to be fully initialized